src/VisualStudio/Core/Def/LanguageService/AbstractLanguageService`2.VsCodeWindowManager.cs

```diff
@@ -234,56 +239,61 @@ public int RemoveAdornments()
 
             // GetOutline is called every time a new code window is created. Whenever we switch to a different window, it is guaranteed
             // that ReleaseOutline will be called on the old window before GetOutline is called for the new window. 
-            int IVsDocOutlineProvider.GetOutline(out IntPtr phwnd, out IOleCommandTarget? ppCmdTarget)
+            int IVsDocOutlineProvider.GetOutline(out IntPtr phwnd, out IOleCommandTarget? pCmdTarget)
             {
+                pCmdTarget = null;
+                GetOutline(out phwnd);
+                return VSConstants.S_OK;
+            }
+
+            private void GetOutline(out IntPtr phwnd)
+            {
+                phwnd = default;
+
                 var enabled = _globalOptions.GetOption(DocumentOutlineOptionsStorage.EnableDocumentOutline);
                 if (!enabled)
-                {
-                    phwnd = default;
-                    ppCmdTarget = null;
-                    return VSConstants.S_OK;
-                }
+                    return;
 
-                var languageServiceBroker = _languageService.Package.ComponentModel.GetService<ILanguageServiceBroker2>();
                 var threadingContext = _languageService.Package.ComponentModel.GetService<IThreadingContext>();
+                threadingContext.ThrowIfNotOnUIThread();
+
+                var languageServiceBroker = _languageService.Package.ComponentModel.GetService<ILanguageServiceBroker2>();
                 var asyncListenerProvider = _languageService.Package.ComponentModel.GetService<IAsynchronousOperationListenerProvider>();
                 var asyncListener = asyncListenerProvider.GetListener(FeatureAttribute.DocumentOutline);
                 var editorAdaptersFactoryService = _languageService.Package.ComponentModel.GetService<IVsEditorAdaptersFactoryService>();
 
-                threadingContext.ThrowIfNotOnUIThread();
-
                 // Assert that the previous Document Outline Control and host have been freed. 
-                Contract.ThrowIfFalse(_documentOutlineControl is null);
+                Contract.ThrowIfFalse(_documentOutlineView is null);
                 Contract.ThrowIfFalse(_documentOutlineViewHost is null);
 
-                _documentOutlineControl = new DocumentOutlineControl(
-                    languageServiceBroker, threadingContext, asyncListener, editorAdaptersFactoryService, _codeWindow);
+                var viewTracker = new VsCodeWindowViewTracker(_codeWindow, threadingContext, editorAdaptersFactoryService);
```

Comment:
Potential follow-up: if we're also creating one of these in navbars, just have the VsCodeWindowManager create one and hold onto it.

Reply:
i was def considering htat. :)